app/scripts/controllers/metametrics.js

```diff
@@ -768,6 +768,8 @@ export default class MetaMetricsController {
       ///: END:ONLY_INCLUDE_IN
       [MetaMetricsUserTrait.SecurityProviders]:
         metamaskState.transactionSecurityCheckEnabled ? ['opensea'] : [],
+      [MetaMetricsUserTrait.SecurityProviders]:
+        metamaskState.securityAlertsEnabled ? ['blockaid'] : [],
```

Comment:
This does not seem right.. You are overwriting the `MetaMetricsUserTrait.SecurityProviders` prop. Meaning that with the current code we will never send the `['opensea']` value.

Also it does not seem compliant with the issue description -> we want to add the `blockaid` to `MetaMetricsUserTrait.SecurityProviders` that was already created for sending `opensea` when opensea provider is enabled. So I was expecting that depending on the state we would actually have to send `['opensea', 'blockaid']` or `['opensea']` or `['blockaid']` or `[]`.

Now this raises a question, I am not sure how segment will handle arrays. Do we need to order them? Is it better to traverse the array and send it in an object format? Or all of that does not matter?

Reply:
Ok so I was syncing with @bschorchit on this and we will not have the current `opensea` and `blockaid` providers being sent simultaneously, but we do want to send the `['blockaid']` in MetaMetricsUserTrait.SecurityProviders, and in the future, when we add new providers, more values should be added to that array.

Reply:
So I added an array that's added to if opensea and blockaid is set. I will remove the opensea implementation in another PR ([issue here](https://github.com/MetaMask/MetaMask-planning/issues/926)), but at least this gives the bases for adding more providers in the future